openapi-core/src/main/java/io/ballerina/openapi/core/GeneratorUtils.java

```diff
@@ -425,27 +424,41 @@ public static OpenAPI getOpenAPIFromOpenAPIV3Parser(Path definitionPath) throws
         return parseResult.getOpenAPI();
     }
 
+    /**
+     * Check whether the given media type is currently supported in the tool.
+     *
+     * @param mediaTypeEntry
+     * @return
+     */
+    public static boolean isSupportedMediaType(Map.Entry<String,
+            io.swagger.v3.oas.models.media.MediaType> mediaTypeEntry) {
+        String mediaType = mediaTypeEntry.getKey();
+        String defaultBallerinaType = getBallerinaMediaType(mediaType, true);
+        if (defaultBallerinaType.equals(HTTP_REQUEST) &&
+                !(mediaTypeEntry.getValue().getSchema() != null && mediaType.equals(MediaType.MULTIPART_FORM_DATA))) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
     /**
      * Generate BallerinaMediaType for all the return mediaTypes.
      */
     public static String getBallerinaMediaType(String mediaType, boolean isRequest) {
-
-        switch (mediaType) {
-            case MediaType.APPLICATION_JSON:
-                return SyntaxKind.JSON_KEYWORD.stringValue();
-            case MediaType.APPLICATION_XML:
-                return SyntaxKind.XML_KEYWORD.stringValue();
-            case MediaType.APPLICATION_FORM_URLENCODED:
-            case MediaType.TEXT_HTML:
-            case MediaType.TEXT_PLAIN:
-                return STRING_KEYWORD.stringValue();
-            case IMAGE_PNG:
-            case MediaType.APPLICATION_OCTET_STREAM:
-            case APPLICATION_PDF:
-            case ANY_TYPE:
-                return isRequest ? HTTP_REQUEST : HTTP_RESPONSE;
-            default:
-                return SyntaxKind.JSON_KEYWORD.stringValue();
+        if (mediaType.equals(MediaType.APPLICATION_JSON) || mediaType.matches("application/.*\\+json") ||
+                mediaType.matches(".*/json")) {
+            return SyntaxKind.JSON_KEYWORD.stringValue();
+        } else if (mediaType.equals(MediaType.APPLICATION_XML) || mediaType.matches("application/.*\\+xml")) {
+            return SyntaxKind.XML_KEYWORD.stringValue();
+        } else if (mediaType.equals(MediaType.APPLICATION_FORM_URLENCODED) || mediaType.equals(MediaType.TEXT_HTML)
+                || mediaType.equals(MediaType.TEXT_PLAIN)) {
```

Comment:
Shall we add regex here to access all the `text/*` related payload types since it is going to map the `string`  type ?

Reply:
In a scenario where the media type is `text/json` what will be the most suitable Ballerina type. + @shafreenAnfar @NipunaRanasinghe 

Ex: 
```bal
 /pets:
    post:
      summary: Create a pet
      operationId: createPet
      requestBody:
        content:
          text/json:
            description: 
      responses:
        '200':
          description: List of existing pets
          content:
            text/json:
              schema:
                $ref: "#/components/schemas/Pets"
```

**Current code generation**

```bal
    remote isolated function createPet(json payload) returns json|error {
        string resourcePath = string `/pets`;
        http:Request request = new;
        request.setPayload(payload, "text/json");
        json response = check self.clientEp->post(resourcePath, request);
        return response;
    }
```

Reply:
@aneeshafedo I prefer having `json` in this case, as this should be referred to a json string. @shafreenAnfar Any thoughts?

Reply:
+1

Reply:
Thank you all